src/workspace.coffee

```diff
@@ -791,36 +799,65 @@ class Workspace extends Model
   # * `regex` {RegExp} to search with.
   # * `options` (optional) {Object} (default: {})
   #   * `paths` An {Array} of glob patterns to search within
+  #   * `onPathsSearched` (optional) {Function}
   # * `iterator` {Function} callback on each file found
   #
-  # Returns a `Promise`.
+  # Returns a *thenable* object with a `cancel()` method that will cancel all
+  # of the underlying searches that were started as part of this scan.
   scan: (regex, options={}, iterator) ->
```

Comment:
No, it now returns a *thenable*, which is an object with a Promise-compatible `then()` method.

A `Promise` has only two instance methods (http://www.html5rocks.com/en/tutorials/es6/promises/#toc-api):

* `then()`
* `cancel()`

I could also support `cancel`, I guess, but the returned object would not pass an `instanceof Promise` test because the code is:

```coffee
return {
  then: promise.then.bind(promise)
  cancel: ->
    isCancelled = true
    directorySearch.cancel()
}
```

In this case, it also supports an undocumented `done()` method, but as you can see in the code, that is only for backwards compatibility with `find-and-replace`.